src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameterSource.cs

```diff
@@ -14,6 +14,7 @@ internal enum EndpointParameterSource
     JsonBodyOrService,
     FormBody,
     Service,
+    KeyedService,
```

Comment:
What happens if someone combines `Service` and `KeyedService`?

Reply:
Good question! I can add a test with an example of the behavior here.

At the present moment, we check for the existence of the `FromService` attribute first which means that it'll always be bound as a non-keyed service.

@benjaminpetit What do you think is the correct behavior when a user annotates a parameter with both the `FromService` and `FromKeyedServices` attributes? I can see two options:

- Throw an exception stating that both attributes are not supported on a parameter
- Check only for keyed services
- Generate some code that resolves from keyed services first and then falls back to non-keyed services

I lean towards Options 1. It strikes me that the two attributes should be mutually exclusive...

Reply:
I also think the first option makes the most sense.